drivers/sensor/bq274xx/bq274xx.c

```diff
@@ -368,6 +368,39 @@ static int bq274xx_sample_fetch(const struct device *dev,
 	return 0;
 }
 
+static int bq274xx_sample_fetch_all(const struct device *dev)
+{
+	int status = 0;
+
+	status = bq274xx_sample_fetch(dev, SENSOR_CHAN_GAUGE_VOLTAGE);
+	status |= bq274xx_sample_fetch(dev, SENSOR_CHAN_GAUGE_AVG_CURRENT);
+	status |= bq274xx_sample_fetch(dev, SENSOR_CHAN_GAUGE_TEMP);
+	status |= bq274xx_sample_fetch(dev, SENSOR_CHAN_GAUGE_STDBY_CURRENT);
+	status |= bq274xx_sample_fetch(dev, SENSOR_CHAN_GAUGE_MAX_LOAD_CURRENT);
+	status |= bq274xx_sample_fetch(dev, SENSOR_CHAN_GAUGE_STATE_OF_CHARGE);
+	status |= bq274xx_sample_fetch(dev, SENSOR_CHAN_GAUGE_FULL_CHARGE_CAPACITY);
+	status |= bq274xx_sample_fetch(dev, SENSOR_CHAN_GAUGE_REMAINING_CHARGE_CAPACITY);
+	status |= bq274xx_sample_fetch(dev, SENSOR_CHAN_GAUGE_NOM_AVAIL_CAPACITY);
+	status |= bq274xx_sample_fetch(dev, SENSOR_CHAN_GAUGE_FULL_AVAIL_CAPACITY);
+	status |= bq274xx_sample_fetch(dev, SENSOR_CHAN_GAUGE_AVG_POWER);
+	status |= bq274xx_sample_fetch(dev, SENSOR_CHAN_GAUGE_STATE_OF_HEALTH);
+
+	if (status < 0) {
+		return -EIO;
+	}
+
+	return 0;
```

Comment:
Because the driver definitely supports these channels, we could do:
```suggestion
	(void)bq274xx_sample_fetch(dev, SENSOR_CHAN_GAUGE_VOLTAGE);
	(void)bq274xx_sample_fetch(dev, SENSOR_CHAN_GAUGE_AVG_CURRENT);
	(void)bq274xx_sample_fetch(dev, SENSOR_CHAN_GAUGE_TEMP);
	(void)bq274xx_sample_fetch(dev, SENSOR_CHAN_GAUGE_STDBY_CURRENT);
	(void)bq274xx_sample_fetch(dev, SENSOR_CHAN_GAUGE_MAX_LOAD_CURRENT);
	(void)bq274xx_sample_fetch(dev, SENSOR_CHAN_GAUGE_STATE_OF_CHARGE);
	(void)bq274xx_sample_fetch(dev, SENSOR_CHAN_GAUGE_FULL_CHARGE_CAPACITY);
	(void)bq274xx_sample_fetch(dev, SENSOR_CHAN_GAUGE_REMAINING_CHARGE_CAPACITY);
	(void)bq274xx_sample_fetch(dev, SENSOR_CHAN_GAUGE_NOM_AVAIL_CAPACITY);
	(void)bq274xx_sample_fetch(dev, SENSOR_CHAN_GAUGE_FULL_AVAIL_CAPACITY);
	(void)bq274xx_sample_fetch(dev, SENSOR_CHAN_GAUGE_AVG_POWER);
	(void)bq274xx_sample_fetch(dev, SENSOR_CHAN_GAUGE_STATE_OF_HEALTH);

	return 0;
```